packetbeat/protos/cassandra/cassandra.go

```diff
@@ -0,0 +1,218 @@
+package cassandra
+
+import (
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+
+	"github.com/elastic/beats/packetbeat/protos"
+	. "github.com/elastic/beats/packetbeat/protos/cassandra/internal/gocql"
+	"github.com/elastic/beats/packetbeat/protos/tcp"
+	"github.com/elastic/beats/packetbeat/publish"
+	"strings"
+)
+
+// cassandra application level protocol analyzer plugin
+type cassandra struct {
+	ports        protos.PortsConfig
+	parserConfig parserConfig
+	transConfig  transactionConfig
+	pub          transPub
+}
+
+// Application Layer tcp stream data to be stored on tcp connection context.
+type connection struct {
+	streams [2]*stream
+	trans   transactions
+}
+
+// Uni-directioal tcp stream state for parsing messages.
+type stream struct {
+	parser parser
+}
+
+var (
+	debugf = logp.MakeDebug("cassandra")
+)
+
+func init() {
+	protos.Register("cassandra", New)
+}
+
+// New create and initializes a new cassandra protocol analyzer instance.
+func New(
+	testMode bool,
+	results publish.Transactions,
+	cfg *common.Config,
+) (protos.Plugin, error) {
+	p := &cassandra{}
+	config := defaultConfig
+	if !testMode {
+		if err := cfg.Unpack(&config); err != nil {
+			return nil, err
+		}
+	}
+
+	if err := p.init(results, &config); err != nil {
+		return nil, err
+	}
+	return p, nil
+}
+
+func (cassandra *cassandra) init(results publish.Transactions, config *cassandraConfig) error {
+	if err := cassandra.setFromConfig(config); err != nil {
+		return err
+	}
+	cassandra.pub.results = results
+	return nil
+}
+
+func (cassandra *cassandra) setFromConfig(config *cassandraConfig) error {
+
+	// set module configuration
+	if err := cassandra.ports.Set(config.Ports); err != nil {
+		return err
+	}
+
+	// set parser configuration
+	parser := &cassandra.parserConfig
+	parser.maxBytes = tcp.TCP_MAX_DATA_IN_STREAM
+
+	// set parser's compressor, only `snappy` supported right now
+	if config.Compressor == Snappy {
+		parser.compressor = SnappyCompressor{}
+	} else {
+		parser.compressor = nil
+	}
+
+	// parsed ignored ops
+	if len(config.OPsIgnored) > 0 {
+		maps := map[FrameOp]bool{}
+
+		if strings.Contains(config.OPsIgnored, ",") {
```

Comment:
sorry, didn't notice it earlier. Do not start splitting strings, but rely on structure provided by configuration support. This is mostly about consistency, as we do this for all fields potentially being an array of strings. If we want to support alternative syntax in future, beats config file reader should have to handle this, but no components should have to care.

Use type `[]string` for `config.OpsIgnored`. This way syntax in yaml file config files becomes

```
ignored_ops:
  - OP1
  - OP2
```

or 

```
ignored_ops: [OP1, OP2]
```

or 

```
ignored_ops: ["OP1", "OP2"]
```
    
```